web/pdf_page_view.js

```diff
@@ -805,41 +781,32 @@ class PDFPageView {
   }
 
   cssTransform({
-    target,
     redrawAnnotationLayer = false,
     redrawAnnotationEditorLayer = false,
     redrawXfaLayer = false,
     redrawTextLayer = false,
     hideTextLayer = false,
   }) {
-    // Scale target (canvas), its wrapper and page container.
-    if (
-      (typeof PDFJSDev === "undefined" || PDFJSDev.test("TESTING")) &&
-      !(target instanceof HTMLCanvasElement)
-    ) {
-      throw new Error("Expected `target` to be a canvas.");
-    }
-    if (!target.hasAttribute("zooming")) {
-      target.setAttribute("zooming", true);
-      const { style } = target;
-      style.width = style.height = "";
+    const { canvas } = this;
+    if (!canvas) {
+      return;
     }
 
-    const originalViewport = this.#viewportMap.get(target);
+    const originalViewport = this.#originalViewport;
     if (this.viewport !== originalViewport) {
       // The canvas may have been originally rotated; rotate relative to that.
       const relativeRotation =
         this.viewport.rotation - originalViewport.rotation;
       const absRotation = Math.abs(relativeRotation);
-      let scaleX = 1,
-        scaleY = 1;
       if (absRotation === 90 || absRotation === 270) {
         const { width, height } = this.viewport;
         // Scale x and y because of the rotation.
-        scaleX = height / width;
-        scaleY = width / height;
+        const scaleX = height / width;
+        const scaleY = width / height;
+        canvas.style.transform = `rotate(${relativeRotation}deg) scale(${scaleX},${scaleY})`;
+      } else if (absRotation === 180) {
+        canvas.style.transform = `rotate(${relativeRotation}deg)`;
       }
```

Comment:
When using CSS-only zooming, which can be enabled by setting the option `maxCanvasPixels = 0`, things seem to break when the page is rotated back to the initial state.
```suggestion
        canvas.style.transform = `rotate(${relativeRotation}deg)`;
      } else {
        canvas.style.transform = ""; // absRotation === 0
      }
      
```